imports/plugins/core/components/lib/hoc.js

```diff
@@ -26,6 +32,56 @@ export function withCurrentUser(component) {
 }
 
 
+/**
+ * @name withMoment
+ * @method
+ * @summary A wrapper to reactively inject the moment package into a component
+ * @param {Function|React.Component} component - the component to wrap
+ * @return {Function} the new wrapped component with a "moment" prop
+ * @memberof Components
+ */
+export function withMoment(component) {
+  return lifecycle({
+    componentDidMount() {
+      try {
+        import("moment").then(moment => {
+          this.setState({
+            moment: moment.default
+          });
```

Comment:
Whenever you call `setState` in a callback, you risk getting the errors that @zenweasel mentioned if the component has already been destroyed before the callback/then got called.

The way I've dealt with this in the past is to set, unset, and check an `isMounted` flag. Some people recommend using a cancelable promise and canceling it when the component dismounts. See answers here: https://stackoverflow.com/questions/39767482/is-there-a-way-to-check-if-the-react-component-is-unmounted

Reply:
Won't this cause the data (`moment`) to be unavailable entirely, if the promise is cancelled when the component is unmounted, if the data isn't loaded yet?